scripts/serie_update.py

```diff
@@ -0,0 +1,590 @@
+        for repo_file in stm32cube_cmsis_templates_path.iterdir():
+            repo_src = stm32cube_cmsis_templates_path / repo_file
+            if repo_src.is_file():
+                os_cmd("cp -r " + str(repo_src)
+                       + " " + str(temp_cmsis_soc_path))
+
+        # for hal and ll drivers
+        temp_drivers_include_path = \
+            self.stm32cube_temp_serie / "drivers" / "include"
+        temp_drivers_include_path.parent.mkdir(parents=True)
+        stm32cube_driver_inc = \
+            self.stm32cube_serie_path / "Drivers" \
+            / Path(self.stm32_seriexx_upper + "_HAL_Driver") / "Inc"
+        os_cmd("cp -r " + str(stm32cube_driver_inc)
+               + "  " + str(temp_drivers_include_path))
+
+        # except for _hal_conf_template.h
+        hal_conf_template = [
+            f
+            for f in temp_drivers_include_path.iterdir()
+            if "hal_conf_template.h" in f.name
+        ][0]
+        hal_conf_template_fullpath = \
+            temp_drivers_include_path / hal_conf_template
+        if hal_conf_template_fullpath.is_file():
+            hal_conf_fullpath = Path(re.sub("_template", r"",
+                                            str(hal_conf_template_fullpath)))
+            if hal_conf_fullpath.exists():
+                hal_conf_fullpath.unlink()
+            hal_conf_template_fullpath.rename(hal_conf_fullpath)
+
+        temp_drivers_src_path = self.stm32cube_temp_serie / "drivers" / "src"
+        temp_drivers_src_path.mkdir()
+        stm32cube_drivers_src_path = \
+            self.stm32cube_serie_path / "Drivers" \
+            / Path(self.stm32_seriexx_upper + "_HAL_Driver") / "Src"
+        os_cmd("cp -r " + str(stm32cube_drivers_src_path) + "/*.* "
+               + str(temp_drivers_src_path))
+
+    ########################################################
+    # Build a commit in temporary dir with STM32Cube version
+    # corresponding to zephyr current hal version
+    #
+    def build_from_current_cube_version(self):
+        # reset the STM32Cube repo to this current version
+        os.chdir(self.stm32cube_serie_path)
+        os_cmd("git reset --hard " + self.current_version)
+
+        # build the zephyr module from the stm32cube
+        self.extract_source()
+        print("Building module from STM32Cube_repo " + self.current_version)
+
+        if not self.stm32cube_temp_serie.parent.exists():
+            self.stm32cube_temp_serie.parent.mkdir(parents=True)
+
+        os.chdir(self.stm32cube_temp)
+        os_cmd("git add -A stm32cube/" + self.stm32_seriexx + "/*")
+        os_cmd('git commit -am "module' + self.current_version + '"')
+
+    # ###################################################
+    # Build patch between zephyr current hal version and
+    # corresponding official STM32Cube version
+    #
+    def build_patch_from_current_zephyr_version(self):
+        # clean-up the module
+        os.chdir(self.stm32cube_temp)
```

Comment:
Avoid changing dir, use absolute paths where necessary

Reply:
I already removed all chdir I can, but some are necessary, specially before a git command (so I move chdir closer to git command), and before removing directories (to avoid being within a directory when it is removed)